x-pack/plugins/security_solution/public/common/components/exceptions/helpers.tsx

```diff
@@ -438,27 +458,43 @@ export const entryHasNonEcsType = (
   }
   for (const { entries } of exceptionItems) {
     for (const exceptionEntry of entries ?? []) {
-      if (indexPatterns.fields.find(({ name }) => name === exceptionEntry.field) === undefined) {
+      if (exceptionEntry.type === 'nested') {
+        for (const nestedExceptionEntry of exceptionEntry.entries) {
+          if (doesFieldNameExist(nestedExceptionEntry) === true) {
+            return true;
+          }
+        }
+      } else if (doesFieldNameExist(exceptionEntry) === true) {
         return true;
       }
     }
   }
   return false;
+
+  function doesFieldNameExist(exceptionEntry: Entry) {
+    return indexPatterns.fields.find(({ name }) => name === exceptionEntry.field) === undefined;
+  }
```

Comment:
Is this named right? It looks like it seeing if there is nothing here through the check against undefined? As if it is saying "doesFieldNameNotExist"?

Some pointers/nits (nothing to hold this up), I would still suggest we follow the flow of most of the code base looking at `== null` rather than something as specific as `undefined` since `== null` will catch both `null` and `undefined` values. Most of the other plugins and core code base typically follow that pattern.


I would also use a `.some` vs. a `.find() === undefined` if that makes more semantic sense within arrays. Most of the times these days I rarely find a reason to use a for loop instead of one of the more semantic array methods such as `.some, .every, .map, .reduce, .flat. etc...`. I still sometimes need to use a `.forEach` or `for of` but I almost always use it as a last resort and try to see if there's a more semantic loop from the array that makes sense first.

Even looking above this at the loops above it, I mentally rewrite it in my mind as a series of `.some()` loops as it seems to be doing a series of loops looking for the first entry to become true to return true but if it cannot find it, then it returns false.

For pure functions, I would be careful of introducing functions within functions. They're harder to isolate and test vs pulling the function out and then testing outside of the original function. Since they're pure functions, I don't really worry about other people calling them if they want to as it doesn't have side effects.

Although we don't have strict linter rules around ES2015+ style fat arrows vs. function names, I would stick to one style within a file when you can and use for example the ES2015+ style.

I still recommend to define the return types within TypeScript for functions as it makes it easier to keep other maintainers from accidentally introducing a new return type by accident.